src/Agent.Listener/Agent.cs

```diff
@@ -103,17 +103,37 @@ public override void Initialize(IHostContext hostContext)
 
                 if (parser.IsCommand("run") && !configManager.IsConfigured())
                 {
-                    throw new InvalidOperationException("Cannot run.  Must configure first.");
+                    throw new InvalidOperationException("Cannot run. Must configure first.");
                 }
 
                 Trace.Info("Done evaluating commands");
+                bool alreadyConfigured = configManager.IsConfigured();
                 await configManager.EnsureConfiguredAsync();
 
                 _inConfigStage = false;
 
-                // TODO: If configured as runasservice return from here
-            
-                return await RunAsync(TokenSource.Token);
+                AgentSettings settings = configManager.LoadSettings();
+                if (parser.IsCommand("run") || !settings.RunAsService)
+                {
+                    // Run the agent interactively
+                    Trace.Verbose(
+                        StringUtil.Format(
+                            "Run command mentioned: {0}, run as service option mentioned:{1}",
+                            parser.IsCommand("run"),
+                            settings.RunAsService));
+
+                    return await RunAsync(TokenSource.Token, settings);
+                }
+
+                if (alreadyConfigured)
+                {
+                    // This is helpful if the user tries to start the agent.listener which is already configured or running as service
+                    // However user can execute the agent by calling the run command
+                    // TODO: Should we check if the service is running and prompt user to start the service if its not already running?
+                    _term.WriteLine(StringUtil.Loc("ConfiguredAsRunAsService", settings.ServiceName));
```

Comment:
:)
When you pass no commands, it does try to configure if its not configured and run the agent.

This behaviour of this when you try to run as service
 - First time configuration 
    Configure
    Exit from here as service will start (no need to print the  last message you service manager already indicates service successfully configured)

- Post configuration if you try to run agent.listener without any command
   - Do not configure as its configured
   - Do not run as its configured to run as service. (But print the message so that user is aware)

- If you try to run with "run" command independent of configuration, just run it as its explicitly mentioned. Note, this is important here. If you dont do this. The service will start the agent.listener with run command and agent.listner will terminate saying it is configured to run as service. So the agent.listener can never be run with out this :)